src/main/java/org/opensearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java

```diff
@@ -236,32 +237,26 @@ private void onGetAnomalyDetectorResponse(GetResponse response) {
         try (XContentParser parser = RestHandlerUtils.createXContentParserFromRegistry(xContentRegistry, response.getSourceAsBytesRef())) {
             ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
             AnomalyDetector existingDetector = AnomalyDetector.parse(parser, response.getId(), response.getVersion());
-            // We have separate flows for realtime and historical detector currently. User
-            // can't change detector from realtime to historical, vice versa.
-            if (existingDetector.isRealTimeDetector() != anomalyDetector.isRealTimeDetector()) {
+            // If detector category field changed, frontend may not be able to render AD result for different detector types correctly.
+            // For example, if detector changed from HC to single entity detector, AD result page may show multiple anomaly
+            // result points on the same time point if there are multiple entities have anomaly results.
+            // If single-category HC changed category field from IP to error type, the AD result page may show both IP and error type
+            // in top N entities list. That's confusing.
+            // So we decide to block updating detector category field.
+            if (!Objects.equals(existingDetector.getCategoryField(), anomalyDetector.getCategoryField())) {
```

Comment:
Seems Object.equals cannot be used here.  Example:

```
jshell> List<String> a = new ArrayList<>()
a ==> []

jshell> a.add("a1")
$2 ==> true

jshell> a.add("a2")
$3 ==> true

jshell> List<String> b = new ArrayList<>()
b ==> []

jshell> b.add("a2")
$5 ==> true

jshell> b.add("a1")
$6 ==> true

jshell> Objects.equals(a, b)
$7 ==> false
```

Reply:
That's by design. Block any changes on category fields including the order to avoid any potential issues like we depends on the order to process differently in future. Any concern if we don't allow user change the category field order?

Reply:
As AD backend doesn't rely on the order of category field, I'm ok to change to comparing list ignoring the order to give some flexibility to API user.